spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -88,6 +88,14 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     sparkSession.emptyDataFrame
   }
 
+  protected def executeVacuum(
+      deltaLog: DeltaLog,
+      retentionHours: Option[Double],
+      dryRun: Boolean): DataFrame = {
+    VacuumCommand.gc(sparkSession, deltaLog, dryRun, retentionHours)
+    sparkSession.emptyDataFrame
```

Comment:
if dry run is going to return a dataframe with list of eligible files, why do we return `emptyDataFrame` here?

Reply:
This is from the existing implementation. Why do these APIs return emptyDataFrame, though vacuum command return the list of data paths? 

Reply:
Do we return list of files when it's run in  NON-DRY run mode, I don't think so. In case of dry run, this is important, isn't it? Did we test that it's returning list of files in spite of empty frame here?

Reply:
https://github.com/delta-io/delta/blob/4f9c8b9cc294ec7b321847115bf87909c356bc5a/spark/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala#L315C16-L315C16

Ok it returns only basePath. Vacuum SQL command returns it for non-dry run mode. 
I can add the implementation. One concern is there's no limit for # of paths to return.

Reply:
@sezruby 
I am seeing many updates to this PR. Let me know once this PR is ready for review again.

Reply:
@rajeshparangi test is green now